src/Language/GraphQL/Draft/Syntax.hs

```diff
@@ -115,15 +124,47 @@ newtype Name = Name {unName :: Text}
   deriving stock (Eq, Lift, Ord, Show)
   deriving newtype (Hashable, NFData, Pretty, Semigroup, J.ToJSONKey, J.ToJSON)
 
+-- | @NameSuffix@ is essentially a GQL identifier that can be used as Suffix
+--  It is slightely different from @Name@ as it relaxes the criteria that a
+--  @Name@ cannot start with a digit.
+type NameSuffix :: Type
+newtype NameSuffix = Suffix {unNameSuffix :: Text}
+  deriving stock (Lift, Show)
+
+matchFirst :: Char -> Bool
+matchFirst c = c == '_' || C.isAsciiUpper c || C.isAsciiLower c
+
+matchBody :: Char -> Bool
+matchBody c = c == '_' || C.isAsciiUpper c || C.isAsciiLower c || C.isDigit c
+
+isValidName :: Text -> Bool
+isValidName text =
```

Comment:
`matchFirst`, `matchBody`, and `isValidName` are all relatively clear from the names, but it would be nice to have a docstring referencing the section of the GraphQ” spec that defines what it means for these functions to be correctly implemented.